### ISSUE TITLE
STIR reconstruction error due to unsupported z-grid spacing

### ISSUE THREAD

**Issue by ckolbPTB** [CCPPETMR/SIRF#330]
I run the example script "reconstruct_from_listmode.py" with

"sirf@sirf-VirtualBox:~/devel/build/sources/SIRF/examples/Python/PET$ python reconstruct_from_listmode.py " without any modifications, so using the provided phantom data. I ran the script 10 times. On a few runs the script finished successfully, but the image I got in the last figure which pops up was zero everywhere. No error message though.

On all other runs I got the same basic error message saying that the image z-spacing does not fit to the ring spaces but always with different values for the image z-spacing. Here are some examples of the error messages:

RUN 1
applying attenuation (please wait, may take a while)...
setting up, please wait...

ERROR: DataSymmetriesForBins_PET_CartesianGrid can currently only support z-grid spacing equal to the ring spacing of the scanner divided by an integer. Sorry. (Image z-spacing is 0.000267 and ring spacing is 4.062500)

RUN 2
applying attenuation (please wait, may take a while)...
setting up, please wait...

ERROR: DataSymmetriesForBins_PET_CartesianGrid can currently only support z-grid spacing equal to the ring spacing of the scanner divided by an integer. Sorry. (Image z-spacing is -0.234735 and ring spacing is 4.062500)

RUN 3
applying attenuation (please wait, may take a while)...
setting up, please wait...

ERROR: DataSymmetriesForBins_PET_CartesianGrid can currently only support z-grid spacing equal to the ring spacing of the scanner divided by an integer. Sorry. (Image z-spacing is -0.000000 and ring spacing is 4.062500)

I guess this means that there is a memory problem somewhere, i.e. that the image z-spacing is reading out a random part of memory...

I am running it on a Ubuntu 18.04 virtual machine using a SIRF_SuperBuild with DDEVEL_BUILD=ON

**Comment by rijobro** [comment]
@paskino, if @ckolbPTB is using the most recent VM, could you test it (might mean there's a bug in the VM). I'll try on my custom VM and we'll see if that's the problem.

**Comment by paskino** [comment]
I've tested it in the newest VM and it works.
https://github.com/CCPPETMR/SIRF/issues/333#issuecomment-473886056

**Comment by rijobro** [comment]
I unfortunately get the same error as Christoph (on a non-official VM). I suppose this hints at it being an inherent SIRF/STIR problem...

**Comment by KrisThielemans** [comment]
can you use this `valgrind` on this? Just
`valgrind python listmode..py`
would tell us something. If not obvious, we'd have to compile with `-g` flags. (not entirely easy in the SuperBuild maybe).

**Comment by evgueni-ovtchinnikov** [comment]
@ckolbPTB, @rijobro:

i would start by printing `vz` in `cSTIR_imageFromAcquisitionDataAndNxNy` (cstir.cpp)

if the value is wrong, next print it (using `get_voxel_sizes`) in `STIRImageData(const PETAcquisitionData& ad)` (stir_data_containers.h)

if it is wrong too, then the bug must be in STIR's `VoxelsOnCartesianGrid` constructor from `ProjDataInfo`

**Comment by rijobro** [comment]
problem should be fixed, just waiting on Travis and then will merge into master.

**Comment by KrisThielemans** [comment]
Fixed by #335
